src/software/power/charger.cpp

```diff
@@ -11,30 +9,20 @@ Charger::Charger()
     pinMode(CHRG_DONE, INPUT);
     pinMode(CHRG, OUTPUT);
 
-    charge_timer = timerBegin(CHARGE_TIMER, 80, true);
-    timerAttachInterrupt(charge_timer, &chargeDone, true);
+    attachInterrupt(digitalPinToInterrupt(CHRG_DONE), &Charger::chargeDone, FALLING);
 }
 
+
 void IRAM_ATTR Charger::chargeDone()
 {
     flyback_fault = digitalRead(FLYBACK_FAULT);
-    if (charge_done_callback)
-    {
-        charge_done_callback();
-        charge_done_callback = NULL;
-    }
-    digitalWrite(CHRG, LOW);
 }
 
 void Charger::chargeCapacitors()
 {
     if (!flyback_fault)
```

Comment:
i'd just do `!digitalRead(FLYBACK_FAULT)` and get rid of the charge done interrupt + function all together